File: ssc/apiwrappers/java/SSC/Info.java
package SSC;

public class Info 
{
    private long m_inf;
    private SSC.Module m_mod;
    private int m_idx;

    public Info(SSC.Module m)
    {
        m_mod = m;
        m_idx = 0;
    }

    public void reset()
    {
        m_idx = 0;
    }

    public boolean get()
    {
        m_inf = SSC.SSCAPIJNI.ssc_module_var_info(m_mod.getModuleHandle(), m_idx);
        if (m_inf == 0)
        {
            reset();
            return false;
        }

        m_idx++;
        return true;
    }

    public String name()
    {
        if (m_inf == 0) return null;
        return SSC.SSCAPIJNI.ssc_info_name(m_inf);
    }

    public int varType()
    {
        if (m_inf == 0) return -1;
        return SSC.SSCAPIJNI.ssc_info_var_type(m_inf);
    }

    public int dataType()
    {
        if (m_inf == 0) return -1;
        return SSC.SSCAPIJNI.ssc_info_data_type(m_inf);
    }

    public String label()
    {
        if (m_inf == 0) return null;
        return SSC.SSCAPIJNI.ssc_info_label(m_inf);
    }

    public String units()
    {
        if (m_inf == 0) return null;
        return SSC.SSCAPIJNI.ssc_info_units(m_inf);
    }

    public String meta()
    {
        if (m_inf == 0) return null;
        return SSC.SSCAPIJNI.ssc_info_meta(m_inf);
    }

    public String group()
    {
        if (m_inf == 0) return null;
        return SSC.SSCAPIJNI.ssc_info_group(m_inf);
    }

    public String required()
    {
        if (m_inf == 0) return null;
        return SSC.SSCAPIJNI.ssc_info_required(m_inf);
    }

    public String constraints()
    {
        if (m_inf == 0) return null;
        return SSC.SSCAPIJNI.ssc_info_constraints(m_inf);
    }
}

File: ssc/apiwrappers/matlab/+SSC/@Info/Info.m
classdef Info < handle
    %Info object from SSC
    %   Object used to gather ssc data meta information  name, label,
    %   units, constraints and other  information.
    
    properties (SetAccess = private)
        m_idx
        m_info
        m_module
    end
    properties
        Name
        VariableType
        DataType
        Label
        Units
        Meta
        Group
        Required
        Constraints
    end
    
    methods
        function obj = Info(module)
            obj.m_idx = 0;
            obj.m_module = module;
        end
        function Reset(obj)
            obj.m_idx = 0;
        end
        function result = Get(obj)
            obj.m_info = SSC.ssccall('module_var_info', obj.m_module.GetHandle(), obj.m_idx);
            if (obj.m_info == 0),
                obj.Reset;
                result = false;
            else
                obj.Name = SSC.ssccall('info_name', obj.m_info);
                obj.VariableType = SSC.ssccall('info_var_type', obj.m_info);
                obj.DataType = SSC.ssccall('info_data_type', obj.m_info);
                obj.Label = SSC.ssccall('info_label', obj.m_info);
                obj.Units = SSC.ssccall('info_units', obj.m_info);
                obj.Meta = SSC.ssccall('info_meta', obj.m_info);
                obj.Group = SSC.ssccall('info_group', obj.m_info);
                obj.Required = SSC.ssccall('info_required', obj.m_info);
                obj.Constraints = SSC.ssccall('info_constraints', obj.m_info);
                obj.m_idx = obj.m_idx + 1;
                result = true;
            end
        end
    end
    
end

